### PR TITLE
Deref pointer when using sizeof in x509_get_other_name

### PR DESCRIPTION
**Pull request by ekse**
Fix for #2716.

### CODE REVIEW

**Comment by ekse** [comment]
I added the ChangeLog entry.

File: ChangeLog [hunk 1]
```diff
@@ -74,6 +74,7 @@ Bugfix
      irwir.
    * Enable Suite B with subset of ECP curves. Make sure the code compiles even
      if some curves are not defined. Fixes #1591 reported by dbedev.
+   * Fix partial zeroing in x509_get_other_name. Fixes #2716.
```
Comment (hanno-arm) [inline ChangeLog]: @ekse If you are OK with it, I'd suggest we add a crediting line like "Found and fixed by YOUR PREFERRED NAME."

Comment (ekse) [reply ChangeLog]: I added credit to the entry.

**Comment by hanno-arm** [comment]
Ping @Patater for gatekeeping.

**Comment by Patater** [comment]
No backports needed because `other_name` and this zeroization is not present in the 2.16 and 2.7 branches.

**Pull request merged**